server/api/room.ts

```diff
@@ -510,6 +492,15 @@ const errorHandler: ErrorRequestHandler = (err: Error, req, res) => {
 					message: err.message,
 				},
 			});
+		} else if (err instanceof ZodError) {
```

Comment:
Pay close attention to the checks being made and where in the control flow they are being made. This check will never get hit where it is now.

Reply:
Ah yep, got it.